contracts/interchain-token-service/src/primitives.rs

```diff
@@ -74,6 +79,36 @@ impl From<DeployInterchainToken> for Message {
     }
 }
 
+#[cw_serde]
+#[derive(Eq)]
+pub struct RegisterToken {
+    pub decimals: u8,
+    pub address: nonempty::HexBinary,
+}
+
+impl From<RegisterToken> for Message {
+    fn from(value: RegisterToken) -> Self {
+        Message::RegisterToken(value)
+    }
+}
+
+#[cw_serde]
+#[derive(Eq)]
+pub struct LinkToken {
+    pub token_id: TokenId,
+    pub token_manager_type: Uint256,
```

Comment:
might be difficult to do because Uint256 is not a rust primitive. Maybe treat it as black box data for now and handle it in more detail if we actually have to differentiate types?

Reply:
We had defined an enum before for it. There's a tradeoff here: the benefit is extra validation but the value itself isn't useful to the hub. The downside of the validation is also that when new token manager types are added to the edge, the Hub has to be updated as well which is additional overhead. For e.g. EVM ITS might add support for some more types, and the other ITS contracts don't really care and will continue accepting only the types they support. So I'm leaning towards a passthrough approach as well